File: all/t220/java/Signature.java
public class UniqueDeque<T> {

    private Deque<T> deque = new LinkedList<>();

    /**
     * Add an item to the deque if it is not already present.
     *
     * @param item The item to add.
     * @return boolean True if the item was added, False if it was already present.
     */
    public boolean add(T item) {
        if (!deque.contains(item)) {
            deque.addLast(item);
            return true;
        }
        return false;
    }

    /**
     * Remove an item from the deque if it exists.
     *
     * @param item The item to remove.
     * @return boolean True if the item was removed, False if it was not found.
     */
    public boolean delete(T item) {
        return deque.remove(item);
    }

    /**
     * Check if an item is present in the deque.
     *
     * @param item The item to check.
     * @return boolean True if the item is present, False otherwise.
     */
    public boolean contains(T item) {
        return deque.contains(item);
    }

    /**
     * Get the number of elements in the deque.
     *
     * @return int The number of unique elements in the deque.
     */
    @Override
    public int size() {
        return deque.size();
    }

    /**
     * Create an iterator for the deque.
     *
     * @return Iterator<T> An iterator over the elements in the deque.
     */
    @Override
    public java.util.Iterator<T> iterator() {
        return deque.iterator();
    }
}

File: all/t220/javascript/signature.js
class UniqueDeque {
    private deque: Deque<any>;

    constructor() {
        this.deque = new Deque();
    }

    /**
     * Add an item to the deque if it is not already present.
     *
     * @param {any} item - The item to add.
     * @returns {boolean} - True if the item was added, False if it was already present.
     */
    public add(item: any): boolean {
        // Implementation here
        return false;
    }

    /**
     * Remove an item from the deque if it exists.
     *
     * @param {any} item - The item to remove.
     * @returns {boolean} - True if the item was removed, False if it was not found.
     */
    public delete(item: any): boolean {
        // Implementation here
        return false;
    }

    /**
     * Check if an item is present in the deque.
     *
     * @param {any} item - The item to check.
     * @returns {boolean} - True if the item is present, False otherwise.
     */
    public contains(item: any): boolean {
        // Implementation here
        return false;
    }

    /**
     * Get the number of elements in the deque.
     *
     * @returns {number} - The number of unique elements in the deque.
     */
    public get length(): number {
        // Implementation here
        return 0;
    }

    /**
     * Create an iterator for the deque.
     *
     * @returns {Iterator<any>} - An iterator over the elements in the deque.
     */
    [Symbol.iterator](): Iterator<any> {
        let index = 0;
        const self = this;
        return {
            next(): IteratorResult<any> {
                if (index < self.deque.length) {
                    return { value: self.deque.toArray()[index++], done: false };
                }
                return { value: undefined, done: true };
            },
        };
    }
}